tests/unit/dogstatsd/test_statsd.py

```diff
@@ -488,6 +488,23 @@ def test_event_constant_tags(self):
             ),
         )
 
+    def test_event_payload_error(self):
+        def func():
+            # define an event payload that is > 8 * 1024
+            message = ["l" for i in range(8 * 1024)]
+            message = "".join(message)
+            payload = {"title": "title", "message": message}
+
+            self.statsd.event(**payload)
+
+        # check that the method fails when the payload is too large
+        with pytest.raises(ValueError):
+            func()
```

Comment:
Why not have the `func()` body inside the `with` here rather than defining it separately from invoking it?

Reply:
I used the `with` statement to match the convention with which `pytest.raises()` is invoked in this script already: on [line 917](https://github.com/DataDog/datadogpy/pull/730/files/80e670ead0cecf2359144bcb78e707fae62701fa#diff-57154b2052d2e9499c5f655cf922894cfa70ceffedbcdfcb326ec52b97cd0cf3L917) and [line 937](https://github.com/DataDog/datadogpy/pull/730/files/80e670ead0cecf2359144bcb78e707fae62701fa#diff-57154b2052d2e9499c5f655cf922894cfa70ceffedbcdfcb326ec52b97cd0cf3L937
).